rosserial_python/src/rosserial_python/SerialClient.py

```diff
@@ -288,13 +286,16 @@ def write(self, data):
         totalsent = 0
 
         while totalsent < length:
-            sent = self.socket.send(data[totalsent:])
+            try:
+                sent = self.socket.send(data[totalsent:])
+            except BrokenPipeError:
+                sent = 0
```

Comment:
Why not `raise RuntimeError(...)` here?

The message "RosSerialServer.write() socket connection broken" below seems inappopriate for the case in which a `BrokenPipeError` was not caught, but zero bytes were sent anyway.

Reply:
Hmm, yeah I wasn't sure what to do here. From https://stackoverflow.com/a/34920917/109517, it seems like `socket.send` can legitimately return zero in at least some non-error scenarios. I suppose the risk is that it would hard loop on this, especially if you had a non-blocking socket, but that's not the case here with a dedicated writer thread. Have updated it to ignore the 0 scenario and instead throw from inside the `except`.